projects/core/src/i18n/i18next/i18next-backend/i18next-http-backend-client.spec.ts

```diff
@@ -5,7 +5,7 @@ import {
   TestRequest,
 } from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
-import { RequestCallback } from 'i18next-http-backend';
+import { RequestCallback } from 'i18next-http-backend/cjs';
```

Comment:
[QUESTION]
Why can't we import types from a primary entry point anymore, but have to use /cjs secondary entry point?

Reply:
RequestCallBack type is not exported by the lib... the ts version used commonjs and they just recently changed it, that's why it was possible before

Reply:
It looks strange at the first sight that we use some typings from `/cjs` secondary entry point of a library, while our code is compiled from and targeted to ESM.

Alternatively, I've tried importing from their `/esm` secondary entry point, but it seems that there the type `RequestCallback` is not exported. Very strange.

I I'm having hard time understanding why the usage of `RequestCallback` is discouraged in ESM context by library authors. I only guess that's an overlook from their side....

What about:
1. remove usage of explicit type and revert using `/cjs` secondary entry point. Instead rely on implicit types inferred from `HttpBackendOptions["request"]` (our custom type: `export type I18nextHttpBackendClient = HttpBackendOptions['request'];`). Note: `HttpBackendOptions` is exported in `i18next-http-backend` :) See proposed change I made on my local:
![image](https://github.com/SAP/spartacus/assets/4001059/de6a3816-5181-4654-ad16-f672f51a9ea0)
Please note the types are implicitly inferred, when I hover over them with a mouse:
![image](https://github.com/SAP/spartacus/assets/4001059/2fa53168-6a8a-4b3d-984b-d0944e180ddd)

Reply:
I don't see it being exported in CJS either. I assume because it's a different module system, it works somehow. 
I tried your change and it breaks the building process. Typescript tries to import the type anyway:
```typescript
export declare const I18NEXT_HTTP_BACKEND_CLIENT: InjectionToken<
  | ((
      options: import('i18next-http-backend').HttpBackendOptions,
      url: string,
      payload: string | {},
      callback: import('i18next-http-backend').RequestCallback
    ) => void)
  | undefined
>;
```

Reply:
Thanks for checking it! Ok, let's go with your solution then (`/cjs`), if it's the only one. 

And in the meantime I suggest to create a ticket in their repo to sort this situation out. Maybe they overlooked exporting those types. Or maybe we're missing something important and we'll learn from it.
Do you think you could create a ticket in their repo?

Reply:
sure I can create a ticket, but I doubt they fix that.